src/playback-watcher.js

```diff
@@ -100,6 +101,16 @@ export default class PlaybackWatcher {
     }
 
     let currentTime = this.tech_.currentTime();
+    let buffered = this.tech_.buffered();
+
+    if (buffered.length && currentTime + 0.1 >= buffered.end(buffered.length - 1)) {
```

Comment:
What do you think of putting this in `waiting_` after the `techWaiting_` checks? It might make sense for either, but I'm inclined to keep it closer to the other logic that cancels out of "waiting" type events and keep this method simple.